### PR TITLE
Feat/cuda wop

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

File: concrete-cuda/cuda/src/bit_extraction.cu [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+              lwe_dimension_in == 8192));
+  // The number of samples should be lower than 4 time the number of streaming
+  // multiprocessors divided by ((k + 1) * l) (the factor 4 being related
+  // to the occupancy of 50%). The only supported value for k is 1, so
+  // k + 1 = 2 for now.
+  int number_of_sm = 0;
+  cudaDeviceGetAttribute(&number_of_sm, cudaDevAttrMultiProcessorCount, 0);
+  assert(("Error (GPU extract bits): the number of input LWEs must be lower or "
+          "equal to the "
+          "number of streaming multiprocessors on the device divided by 8 * "
```
Comment (pdroalves) [inline concrete-cuda/cuda/src/bit_extraction.cu:30]: This phrase doesn't match with the equation.

**Review by pdroalves — CHANGES_REQUESTED** [review]

File: concrete-cuda/cuda/src/bit_extraction.cu (hunk 1 above)
Comment (pdroalves) [reply concrete-cuda/cuda/src/bit_extraction.cu:30]: I believe the mistake is in the equation, that should be 

```
number_of_samples <= number_of_sm * 4. / 2. / level_count_bsk
```

as in https://github.com/zama-ai/concrete-core/blob/c3a2b083c2e8fc2226e2730424783cf5595dce9a/concrete-cuda/cuda/src/wop_bootstrap.cu (line 28).

Do you agree @bbarbakadze ?

**Comment by pdroalves** [comment]
@slab-ci cpu_test
@slab-ci gpu_test

File: concrete-cuda/cuda/src/bit_extraction.cu (hunk 1 above)
Comment (pdroalves) [reply concrete-cuda/cuda/src/bit_extraction.cu:30]: @agnesLeroy I've modified the formula to match the description. Let's double check it together when you are back.

**Review by pdroalves — APPROVED** [review]

**Pull request merged**